cmake/templates/BoostConfig.cmake.in

```diff
@@ -102,3 +102,12 @@ if(TARGET Boost::iostreams)
   include("${CMAKE_CURRENT_LIST_DIR}/BoostZLIBDependency.cmake" OPTIONAL)
   include("${CMAKE_CURRENT_LIST_DIR}/BoostBZip2Dependency.cmake" OPTIONAL)
 endif()
+
+if(TARGET Boost::thread)
+  find_package(Threads)
```

Comment:
If we are using it without `REQUIRED` then there must be check `if(*_FOUND)` somewhere? It will be easier to add `REQUIRED` though.